File: leetcode/150-逆波兰表达式求值/150-逆波兰表达式求值.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens) {
    const stack = [];
    const n = tokens.length;
    for (let i = 0; i < n; i++) {
        if (tokens[i] === "+" || tokens[i] === "*" || tokens[i] === "-" || tokens[i] === "/") {
            const num1 = stack.pop();
            const num2 = stack.pop();
            const operator = tokens[i];
            if (operator === "*") {
                stack.push(num1 * num2);
            } else if (operator === "+") {
                stack.push(num1 + num2);
            } else if (operator === "-") {
                stack.push(num2 - num1);
            } else if (operator === "/") {
                stack.push(num2 / num1 > 0 ? Math.floor(num2 / num1) : Math.ceil(num2 / num1));
            }
        } else {
            stack.push(parseInt(tokens[i]));
        }
    }
    return stack.pop();
};

File: leetcode/150-逆波兰表达式求值/150-逆波兰表达式求值.py
# 150-逆波兰表达式求值
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i == "+" or i == "-" or i == "*" or i == "/":
                n = len(stack)
                num1,num2 = stack[-1],stack[-2]
                stack = stack[:n-2]
                if i == "+":
                    stack.append(num1 + num2)
                elif i == "-":
                    stack.append(num2 - num1)
                elif i == "*":
                    stack.append(num1 * num2)
                elif i == "/":
                    stack.append(int(num2 / num1))
            else:
                stack.append(int(i))
            
        
        return stack[-1]